bluebutton-data-pipeline-rif-extract/src/test/java/gov/hhs/cms/bluebutton/datapipeline/rif/extract/s3/task/ManifestEntryDownloadTaskIT.java

```diff
@@ -38,29 +49,41 @@ public ManifestEntryDownloadTaskIT() {
 	@Test
 	public void testMD5ChkSum()
 			throws Exception {
+		ExtractionOptions options = new ExtractionOptions(String.format("bb-test-%d", new Random().nextInt(1000)));
+		AmazonS3 s3Client = S3Utilities.createS3Client(options);
+		Bucket bucket = null;
 		try {
-			GetObjectRequest objectRequest = new GetObjectRequest("pdcws301.etl",
-					String.format("%s/%s/%s", "Sample", "2017-10-21T11:48:29Z", "manifest_manifest.xml"));
+			bucket = s3Client.createBucket(options.getS3BucketName());
+			LOGGER.info("Bucket created: '{}:{}'", s3Client.getS3AccountOwner().getDisplayName(), bucket.getName());
+			DataSetManifest manifest = new DataSetManifest(Instant.now(), 0,
+					new DataSetManifestEntry("beneficiaries.rif", RifFileType.BENEFICIARY));
+
+			// upload beneficiary sample file to S3 bucket created above
+			s3Client.putObject(DataSetTestUtilities.createPutRequest(bucket, manifest));
+			s3Client.putObject(DataSetTestUtilities.createPutRequest(bucket, manifest, manifest.getEntries().get(0),
+					StaticRifResource.SAMPLE_A_BENES.getResourceUrl()));
+
+			// download file from S3 that was just uploaded above
+			GetObjectRequest objectRequest = new GetObjectRequest(bucket.getName(),
+					String.format("%s/%s/%s", DataSetMonitorWorker.S3_PREFIX_PENDING_DATA_SETS,
+							manifest.getEntries().get(0).getParentManifest().getTimestampText(),
+							manifest.getEntries().get(0).getName()));
 			Path localTempFile = Files.createTempFile("data-pipeline-s3-temp", ".rif");
-			s3TaskManager = new S3TaskManager(new ExtractionOptions("pdcws301.etl"));
+			s3TaskManager = new S3TaskManager(new ExtractionOptions(options.getS3BucketName()));
 			LOGGER.info("Downloading '{}' to '{}'...", objectRequest.getKey(),
 					localTempFile.toAbsolutePath().toString());
 			Download downloadHandle = s3TaskManager.getS3TransferManager().download(objectRequest,
 							localTempFile.toFile());
 			downloadHandle.getProgress();
 			downloadHandle.waitForCompletion();
 
-			LOGGER.info("localTempFile is  " + localTempFile.toString());
-			ManifestEntryDownloadTask manifestEntryDownloadTask = new ManifestEntryDownloadTask(s3TaskManager, null,
-					null, null);
-			String generatedMD5ChkSum = manifestEntryDownloadTask.
-					getMD5ChkSum(localTempFile);
+			String generatedMD5ChkSum = ManifestEntryDownloadTask.computeMD5ChkSum(localTempFile);
 			LOGGER.info("The generated MD5 value from Java (Base64 encoded) is:" + generatedMD5ChkSum);
 
 			String downloadedFileMD5ChkSum = downloadHandle.getObjectMetadata().getUserMetaDataOf("md5chksum");
 			LOGGER.info("The MD5 value from AWS S3 file's metadata is: " + downloadedFileMD5ChkSum);
-
-			if (!generatedMD5ChkSum.equals(downloadedFileMD5ChkSum))
+			// TODO Remove null check below once Jira CBBD-368 is completed
+			if ((downloadedFileMD5ChkSum != null) && (!generatedMD5ChkSum.equals(downloadedFileMD5ChkSum)))
```

Comment:
Is the new (first) `if` clause here needed. You're creating the S3 objects as part of this test case, right? So they're guaranteed to have a checksum?